src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/AbstractSyntaxFactsService.cs

```diff
@@ -488,12 +488,47 @@ public bool SpansPreprocessorDirective(IEnumerable<SyntaxToken> tokens)
         private bool SpansPreprocessorDirective(SyntaxTriviaList list)
             => list.Any(t => IsPreprocessorDirective(t));
 
-        public bool IsInHeader(SyntaxNode node, SyntaxNode ownerOfHeader, SyntaxNodeOrToken lastTokenOrNodeOfHeader)
+        public bool IsOnHeader(int position, SyntaxNode ownerOfHeader, SyntaxNodeOrToken lastTokenOrNodeOfHeader)
         {
             var start = GetStartOfNodeExcludingAttributes(ownerOfHeader);
             var end = lastTokenOrNodeOfHeader.FullSpan.End;
 
-            return node.Span.Start >= start && node.Span.End <= end;
+            return start <= position && position <= end;
+        }
```

Comment:
small nit (and it's really up to you).  But i would personally prefer that this helper just be implemented as: `return IsInHeader(position, owner, last, ImmutableArray.Empty);`

That way, all the 'meat' of the code is in one location, and this is just a helper.  You could extend things further by just having the core method, and making this into an extension on ISyntaxFactsService.

Reply:
nit:  consider:

```c#
var headerSpan = Span.FromBound(GetStart..., lastNodeOrToken...);
return headerSpan.IntersectsWith(position);
```

I personally find the `.IntersectsWith(position)` easier to reason about rather than having to figure out that expression.  up to you.

Reply:
1) Yep, that makes more sense. 

2) I'd agree but (especially if we do 1)) this way it contrasts better with the "holes" check that is exclusive. And while this way it might be _slightly_ less readable it provides context to holes check which makes it a slight net positive. 

There could be an argument for an extension method that is InterserctsWithExclusive or smth (not sure where that would live) which would solve it, tho.

Reply:
'IntersectsWith' is already exclusive :-)

Reply:
IntersectsWith is inclusive but for holes I need exclusive. I.e. yes, I can do `return headerSpan.IntersectsWith(position);` but I can't do `IntersectsWith` on holes.

Reply:
I have a fix for that, tho.